src/printer.js

```diff
@@ -130,6 +130,107 @@ function printLines(path, options, print) {
   return join(hardline, printed);
 }
 
+// special layout for "chained" function calls, i.e.
+// $foo = a()
+//     ->b()
+//     ->c();
+function printMemberChain(path, options, print) {
+  // First step: Linearize and reorder the AST.
+  //
+  // Example:
+  //   a()->b->c()->d()
+  // has the AST structure
+  //   Call (PropertyLookup d (
+  //     Call (PropertyLookup c (
+  //       PropertyLookup b (
+  //         Call (Identifier a)
+  //       )
+  //     ))
+  //   ))
+  // and we transform it into (notice the reversed order)
+  //   [Identifier a, Call, PropertyLookup b, PropertyLookup c, Call,
+  //    PropertyLookup d, Call]
+  const printedNodes = [];
+
+  // recursive call to traverse AST
+  function rec(path) {
```

Comment:
maybe give this a more semantic name?

Reply:
i thought the same when i copied it over from the JS printer, and added the comment instead :smile: I'll rename it and remove the comment. How about `traverse`?

Reply:
sounds great!